profile/prune.go

```diff
@@ -37,6 +37,10 @@ func (p *Profile) Prune(dropRx, keepRx *regexp.Regexp) {
 				funcName := strings.TrimPrefix(fn.Name, ".")
 				// Account for unsimplified names -- trim starting from the first '('.
 				if index := strings.Index(funcName, "("); index > 0 {
+					// ... unless the thing in brackets is an anonymous namespace marker
+					if !strings.HasPrefix(funcName[index:], "(anonymous namespace)") {
+						funcName = funcName[:index]
```

Comment:
Is it intentional that "funcName = funcName[:index] will execute twice (here and two lines below) if there is no prefix match?